capnpc/src/codegen.rs

```diff
@@ -1409,29 +1409,16 @@ fn generate_pipeline_getter(
     let name = get_field_name(field)?;
 
     match field.which()? {
-        field::Group(group) => {
-            let params = get_params(gen, group.get_type_id())?;
-            let params_string = if params.is_empty() {
-                "".to_string()
-            } else {
-                format!("<{}>", params.join(","))
-            };
-
-            let the_mod = gen.scope_map[&group.get_type_id()].join("::");
-            Ok(Branch(vec![
-                Line(format!(
-                    "pub fn get_{}(&self) -> {}::Pipeline{} {{",
-                    camel_to_snake_case(name),
-                    the_mod,
-                    params_string
-                )),
-                Indent(Box::new(Line(
-                    "::capnp::capability::FromTypelessPipeline::new(self._typeless.noop())"
-                        .to_string(),
-                ))),
-                Line("}".to_string()),
-            ]))
-        }
+        field::Group(_group) => Ok(Branch(vec![
+            Line(format!(
+                "pub fn get_{}(&self) -> Self {{",
```

Comment:
I think this generates incorrect code. I pushed a commit that adds coverage to show what I mean: https://github.com/capnproto/capnproto-rust/commit/8ead5d3a96c8da738a19092a849b4ab18ea9eb6f

Reply:
hmm, looking at it again i'm not super confident.

What does your new code do? I'll rebase and see what happens

Reply:
`check` and `test` still pass for me locally after rebase. I'm struggling to convince myself that I'm reasoning about this code correctly, so I'm relying on you and CI here.
Do you think there's coverage that would fail if this was *not* generating valid code?

Reply:
i don't think i closed this PR? Or i certainly didn't mean to. Don't seem to have perms to reopen it

Reply:
```
$ cargo test --all
   Compiling capnp-rpc-test v0.0.0 (/home/dwrensha/src/capnproto-rust/capnp-rpc/test)
   Compiling capnpc-test v0.0.0 (/home/dwrensha/src/capnproto-rust/capnpc/test)
error[E0599]: no method named `get_cap_in_group` found for struct `box_::Pipeline` in the current scope
    --> capnp-rpc/test/test.rs:291:58
     |
291  |         let _ = promise.pipeline.get_out_box().get_foo().get_cap_in_group();
     |                                                          ^^^^^^^^^^^^^^^^ method not found in `box_::Pipeline`
     |
    ::: /home/dwrensha/src/capnproto-rust/target/debug/build/capnp-rpc-test-2c91d10090413622/out/test_capnp.rs:4259:5
     |
4259 |     pub struct Pipeline { _typeless: ::capnp::any_pointer::Pipeline }
     |     ------------------- method `get_cap_in_group` not found for this struct

For more information about this error, try `rustc --explain E0599`.
error: could not compile `capnp-rpc-test` due to previous error
warning: build failed, waiting for other jobs to finish...
```